src/frameobject.ts

```diff
@@ -133,6 +135,11 @@ class Py_FrameObject implements IPy_FrameObj {
                 console.log(`${t.stackDepth()}: ${opcodes[op]}`);
             }
             func(this, t);
+            // Pop the generator frame (whenever a YIELD_VALUE occurs)
+            if(this.returnToThread && this.genFrame){
+                t.framePop();
```

Comment:
Meaning: I don't think the `genFrame` boolean or this special case is even needed.